File: yongkeun/1week/약수구하기.js
function solution(n) {
  var answer = [];
  for (let i = 1; i <= n; i++) {
    //i를 1부터 n까지 반복문 실행
    if (n % i == 0) {
      //n과 i를 나눴을 때 나머지가 0이면 약수이다. 그럴때 if문 실행
      answer.push(i); //약수라면 answer이라는 배열에 넣음
    }
  }
  return answer;
}

File: minwook/약수_구하기.java
import java.util.*;

class Solution {
    public int[] solution(int n) {
        List<Integer> answerList = new ArrayList<>(); // 동적 배열 리스트 선언
        for (int i = 1; i < n + 1; i++) { // 리스트에 약수 넣기
            if (n % i == 0) {
                answerList.add(i);
            }
        }
        
        int size = answerList.size(); // 리스트 크기 선언
        int[] answer = new int[size]; // 정적 배열 선언
        for (int i = 0; i < size; i++) { // 리스트 값 -> 정적 배열
            answer[i] = answerList.get(i);
        }
        return answer; // 정적 배열 리턴
    }
}